packages/editor/src/bindings/post-meta.js

```diff
@@ -25,4 +25,25 @@ export default {
 			.getEditedEntityRecord( 'postType', postType, context.postId )
 			.meta?.[ args.key ];
 	},
+	setValue( { registry, context, args, value } ) {
+		const postType = context.postType
+			? context.postType
+			: registry.select( editorStore ).getCurrentPostType();
+		registry
+			.dispatch( coreDataStore )
+			.editEntityRecord( 'postType', postType, context.postId, {
+				meta: {
+					[ args.key ]: value,
+				},
+			} );
+	},
+	lockAttributesEditing() {
+		// TODO: Check that editing is happening in the post editor and not a template.
+
+		// TODO: Check that the custom field is not protected and available in the REST API.
+
+		// TODO: Check that the user has the capability to edit post meta.
```

Comment:
@cbravobernal explored the possibility of creating a new selector to check if the user is an admin [here](https://github.com/WordPress/gutenberg/pull/61734/commits/9659455eed4343c455760b2f9076fe3a5bdbf2d7). We also discussed the possibility of using [canUserEditEntityRecord](https://github.com/WordPress/gutenberg/blob/9659455eed4343c455760b2f9076fe3a5bdbf2d7/packages/core-data/src/selectors.ts#L1164).